bin/BuildPackages.sh

```diff
@@ -21,55 +50,82 @@ set -e
 # an idea what to do for a complete installation of GAP.
 
 
-if [ $# -eq 0 ]
-  then
-    echo "Assuming default GAP location: ../.."
-    GAPDIR=../..
-  else
-    echo "Using GAP location: $1"
-    GAPDIR="$1"
+# Is someone trying to run us from inside the 'bin' directory?
+if [[ -f gapicon.bmp ]]
+then
+  error "This script must be run from inside the pkg directory" \
+        "Type: cd ../pkg; ../bin/BuildPackages.sh"
 fi
 
-# Is someone trying to run us from inside the 'bin' directory?
-if [ -f gapicon.bmp ]
-  then
-    echo "This script must be run from inside the pkg directory"
-    echo "Type: cd ../pkg; ../bin/BuildPackages.sh"
-    exit 1
+# CURDIR
+CURDIR="$(pwd)"
+
+# GAPDIR
+if [[ "$#" -ge 1 ]]
+then
+  case "$1" in
+    --with-gaproot=*)
+      GAPDIR="$(sed 's|--with-gaproot=||' <<< $1)"
+      shift
+    ;;
+
+    *)
+      cd ..
+      GAPDIR="$(pwd)"
+      cd "$CURDIR"
+    ;;
```

Comment:
I don't understand: So if I pass in any argument that is *not* of the form `---with-gaproot=*`, then that argument is ignored, and you set GAPDIR to the parent directory of the current dir? So if I enter the (previously valid) command sequence
```
cd /my/custom/pkg
$GAPDIR/bin/BuildPackages.sh $GAPDIR
```
it will now break?!